akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -362,6 +363,17 @@ private[projection] object R2dbcProjectionImpl {
                 Some(loadedEnvelope)
               }
             } else {
+              // FIXME do we need the same for all projection handler variants?
+              // FIXME not accepted may be that a producer filter decided to start including events,
+              //       how do we know for sure and what offset to start from?
+              env match {
+                case ee: EventEnvelope[_] if ee.sequenceNr > 0 =>
+                  log.debug("Triggering replay for unexpected offset envelope: [{}]", env)
+                  ConsumerFilter(system).ref ! ConsumerFilter.Replay(
+                    "producer-filter-e2e-test", // FIXME where do we get stream id from?
```

Comment:
yeah, my idea was

> The R2dbcProjectionImpl can trigger a ReplayReq via the ConsumerFilter extension if the R2dbcOffsetStore rejects the envelope. In that case it would not fail the stream. The translation from entityType to streamId can be done by the FilterStage since it can look at the entityType of the first received envelope to understand the mapping between entityType and streamId.

Somehow we need to get hold of the ConsumerFilter extension. We anyway have an optional dependency from projection-r2dbc to projection-grpc so maybe we can optionally "find" the ConsumerFilter extension. This replay mechanism should only be used when the projection is used together with projection-grpc.

Here we only know the entityType, so we need to trigger a replay based on the entityType, and let the ConsumerFilter handle the translation to streamId, as I mentioned above.